client/client.go

```diff
@@ -265,6 +296,29 @@ func (c *Client) NewUploadRequest(ctx context.Context, method, urlStr string, bo
 	return req, nil
 }
 
+// NewUploadRequest handles image uploads for image service
+func (c *Client) NewUnAuthUploadRequest(ctx context.Context, method, urlStr string, body []byte) (*http.Request, error) {
```

Comment:
I feel we can reuse a lot of code based on conditions. You may try having builder pattern for request creation. We can take it as a separate improvement.